multiplex.py

```diff
@@ -258,6 +261,54 @@ def convert_vals(obj):
 
     return new_obj
 
+def load_presets(json_req):
+    """Create a dict for essentials, defaults and includable presets"""
+    if bool(json_req) and "presets" in json_req:
+        # copy the essentials and defaults to a dict
+        # to make our lives easier when overriding params
+        if "essentials" in json_req["presets"]:
+            for _ess in json_req["presets"]["essentials"]:
+                essentials_dict.update({ _ess["arg"]: _ess["vals"] })
+        if "defaults" in json_req["presets"]:
+            for _def in json_req["presets"]["defaults"]:
+                defaults_dict.update({ _def["arg"]: _def["vals"] })
+
+        # save all "includables" presets
+        for _preset in json_req["presets"]:
+            if _preset != "defaults" and _preset != "essentials":
+                presets_dict.update({ _preset: json_req["presets"][_preset] })
+
+def override_presets(json_obj):
+    """Override params w/ presets loaded from the requirements file"""
+    for _json in json_obj:
+        _defaults = copy.deepcopy(defaults_dict)
+        # step 1: delete from defaults all params defined
+        for _param in _json:
+            if _param["arg"] in _defaults:
+                # delete param so we append only undefined defaults
+                del _defaults[_param["arg"]]
+        # step 2: append to the params the defaults presets
+        # append default presets (undefined params)
+        for _def in _defaults:
+            _json.append({ "arg": _def, "vals": _defaults[_def] })
```

Comment:
This should only happen if `_json` is empty, correct?

Reply:
No. The params that are defined in the json and also in the deafults, we delete them from the _defaults dict. What will last at the end ? Only the defaults that are not defined, so we just append those to the json.
In other words, we check what defaults are defined, and add the undefined dedfaults.